diff -urN mailman-2.1.5.org/cron/mailpasswds mailman-2.1.5/cron/mailpasswds
--- mailman-2.1.5.org/cron/mailpasswds	2004-07-03 20:38:23.423133536 +0200
+++ mailman-2.1.5/cron/mailpasswds	2004-07-03 21:16:28.111808064 +0200
@@ -208,6 +208,7 @@
                  'owner'   : siteowner,
                  }, lang=poplang)
             # Coerce everything to Unicode
+            header = tounicode(header, enc)
             text = tounicode(text, enc)
             table = [tounicode(_t, enc) for _t in table]
             # Add the table to the end so it doesn't get wrapped/filled
