--- mailman-2.1.9/Mailman/Gui/General.py	2006-08-30 17:54:22.000000000 +0300
+++ mailman-2.1.9/Mailman/Gui/General.py	2007-09-20 17:45:31.857725007 +0300
@@ -409,6 +409,13 @@
              headers.)"""))
             )
 
+        # add X-IMAP-Folder header
+        rtn.append(
+            ('x_imap_folder', mm_cfg.String, WIDTH, 0,
+             _('''If filled add to each outgroing email X-IMAP-Folder header
+             with this value.'''))
+            )
+
         # Discard held messages after this number of days
         rtn.append(
             ('max_days_to_hold', mm_cfg.Number, 7, 0,
--- mailman-2.1.20/Mailman/Handlers/CookHeaders.py~	2015-03-31 20:21:49.000000000 +0300
+++ mailman-2.1.20/Mailman/Handlers/CookHeaders.py	2015-05-25 11:28:53.400847919 +0300
@@ -107,6 +107,11 @@
     # X-BeenThere's intact.
     change_header('X-BeenThere', mlist.GetListEmail(),
                   mlist, msg, msgdata, delete=False)
+
+    # X-IMAP-Folder header, used commonly for server side filtering to folders.
+    if mlist.x_imap_folder:
+        msg['X-IMAP-Folder'] = mlist.x_imap_folder
+
     # Add Precedence: and other useful headers.  None of these are standard
     # and finding information on some of them are fairly difficult.  Some are
     # just common practice, and we'll add more here as they become necessary.
--- mailman-2.1.14/Mailman/MailList.py~	2010-09-20 21:18:27.000000000 +0300
+++ mailman-2.1.14/Mailman/MailList.py	2010-09-21 18:33:08.352249806 +0300
@@ -363,6 +363,7 @@
         self.available_languages = []
         self.include_rfc2369_headers = 1
         self.include_list_post_header = 1
+        self.x_imap_folder = ''
         self.include_sender_header = 1
         self.filter_mime_types = mm_cfg.DEFAULT_FILTER_MIME_TYPES
         self.pass_mime_types = mm_cfg.DEFAULT_PASS_MIME_TYPES
--- mailman-2.1.9/Mailman/versions.py	2005-12-30 20:50:08.000000000 +0200
+++ mailman-2.1.9/Mailman/versions.py	2007-09-20 16:49:02.206553877 +0300
@@ -407,6 +407,8 @@
     add_only_if_missing('collapse_alternatives',
                         mm_cfg.DEFAULT_COLLAPSE_ALTERNATIVES)
 
+    add_only_if_missing('x_imap_folder', '')
+
 
 
 def UpdateOldUsers(mlist):
