diff -Naur mailman-2.1.11.orig/Mailman/Errors.py mailman-2.1.11/Mailman/Errors.py
--- mailman-2.1.11.orig/Mailman/Errors.py	2008-06-30 18:29:46.000000000 +0200
+++ mailman-2.1.11/Mailman/Errors.py	2008-11-12 16:46:05.000000000 +0100
@@ -50,13 +50,13 @@
 class MMInvalidCookieError(MMCookieError): pass
 
 # BAW: these still need to be converted to classes.
-MMMustDigestError    = "MMMustDigestError"
-MMCantDigestError    = "MMCantDigestError"
-MMNeedApproval       = "MMNeedApproval"
-MMSubscribeNeedsConfirmation = "MMSubscribeNeedsConfirmation"
-MMBadConfirmation    = "MMBadConfirmation"
-MMAlreadyDigested    = "MMAlreadyDigested"
-MMAlreadyUndigested  = "MMAlreadyUndigested"
+class MMMustDigestError(Exception): pass
+class MMCantDigestError(Exception): pass
+class MMNeedApproval(Exception): pass
+class MMSubscribeNeedsConfirmation(Exception): pass
+class MMBadConfirmation(Exception): pass
+class MMAlreadyDigested(Exception): pass
+class MMAlreadyUndigested(Exception): pass
 
 MODERATED_LIST_MSG    = "Moderated list"
 IMPLICIT_DEST_MSG     = "Implicit destination"
