diff -urNp -x '*.orig' mailman-2.1.34.org/cron/bumpdigests mailman-2.1.34/cron/bumpdigests
--- mailman-2.1.34.org/cron/bumpdigests	2020-06-27 02:21:21.000000000 +0200
+++ mailman-2.1.34/cron/bumpdigests	2021-10-23 13:32:45.963703728 +0200
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@ -S
 #
 # Copyright (C) 1998-2018 by the Free Software Foundation, Inc.
 #
diff -urNp -x '*.orig' mailman-2.1.34.org/cron/checkdbs mailman-2.1.34/cron/checkdbs
--- mailman-2.1.34.org/cron/checkdbs	2020-06-27 02:21:21.000000000 +0200
+++ mailman-2.1.34/cron/checkdbs	2021-10-23 13:32:45.963703728 +0200
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@ -S
 #
 # Copyright (C) 1998-2018 by the Free Software Foundation, Inc.
 #
diff -urNp -x '*.orig' mailman-2.1.34.org/cron/crontab.in.in mailman-2.1.34/cron/crontab.in.in
--- mailman-2.1.34.org/cron/crontab.in.in	2020-06-27 02:21:21.000000000 +0200
+++ mailman-2.1.34/cron/crontab.in.in	2021-10-23 13:32:45.963703728 +0200
@@ -1,27 +1,27 @@
 # At 8AM every day, mail reminders to admins as to pending requests.
 # They are less likely to ignore these reminders if they're mailed
 # early in the morning, but of course, this is local time... ;)
-0 8 * * * @PYTHON@ -S @prefix@/cron/checkdbs
+0 8 * * * @MAILMAN_USER@ @prefix@/cron/checkdbs
 #
 # At 9AM, send notifications to disabled members that are due to be
 # reminded to re-enable their accounts.
-0 9 * * * @PYTHON@ -S @prefix@/cron/disabled
+0 9 * * * @MAILMAN_USER@ @prefix@/cron/disabled
 #
 # Noon, mail digests for lists that do periodic as well as threshhold delivery.
-0 12 * * * @PYTHON@ -S @prefix@/cron/senddigests
+0 12 * * * @MAILMAN_USER@ @prefix@/cron/senddigests
 #
 # 5 AM on the first of each month, mail out password reminders.
-0 5 1 * * @PYTHON@ -S @prefix@/cron/mailpasswds
+0 5 1 * * @MAILMAN_USER@ @prefix@/cron/mailpasswds
 #
 # Every 5 mins, try to gate news to mail.  You can comment this one out
 # if you don't want to allow gating, or don't have any going on right now,
 # or want to exclusively use a callback strategy instead of polling.
-0,5,10,15,20,25,30,35,40,45,50,55 * * * * @PYTHON@ -S @prefix@/cron/gate_news
+0,5,10,15,20,25,30,35,40,45,50,55 * * * * @MAILMAN_USER@ @prefix@/cron/gate_news
 #
 # At 3:27am every night, regenerate the gzip'd archive file.  Only
 # turn this on if the internal archiver is used and
 # GZIP_ARCHIVE_TXT_FILES is false in mm_cfg.py
-27 3 * * * @PYTHON@ -S @prefix@/cron/nightly_gzip
+27 3 * * * @MAILMAN_USER@ @prefix@/cron/nightly_gzip
 #
 # At 4:30AM daily, cull old entries from the 'bad' and 'shunt' queues.
-30 4 * * * @PYTHON@ -S @prefix@/cron/cull_bad_shunt
+30 4 * * * @MAILMAN_USER@ @prefix@/cron/cull_bad_shunt
diff -urNp -x '*.orig' mailman-2.1.34.org/cron/disabled mailman-2.1.34/cron/disabled
--- mailman-2.1.34.org/cron/disabled	2020-06-27 02:21:21.000000000 +0200
+++ mailman-2.1.34/cron/disabled	2021-10-23 13:32:45.963703728 +0200
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@ -S
 #
 # Copyright (C) 2001-2018 by the Free Software Foundation, Inc.
 #
diff -urNp -x '*.orig' mailman-2.1.34.org/cron/gate_news mailman-2.1.34/cron/gate_news
--- mailman-2.1.34.org/cron/gate_news	2020-06-27 02:21:21.000000000 +0200
+++ mailman-2.1.34/cron/gate_news	2021-10-23 13:32:45.963703728 +0200
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@ -S
 #
 # Copyright (C) 1998-2018 by the Free Software Foundation, Inc.
 #
diff -urNp -x '*.orig' mailman-2.1.34.org/cron/mailpasswds mailman-2.1.34/cron/mailpasswds
--- mailman-2.1.34.org/cron/mailpasswds	2020-06-27 02:21:21.000000000 +0200
+++ mailman-2.1.34/cron/mailpasswds	2021-10-23 13:32:45.963703728 +0200
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@ -S
 #
 # Copyright (C) 1998-2018 by the Free Software Foundation, Inc.
 #
diff -urNp -x '*.orig' mailman-2.1.34.org/cron/nightly_gzip mailman-2.1.34/cron/nightly_gzip
--- mailman-2.1.34.org/cron/nightly_gzip	2020-06-27 02:21:21.000000000 +0200
+++ mailman-2.1.34/cron/nightly_gzip	2021-10-23 13:32:45.963703728 +0200
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@ -S
 # 
 # Copyright (C) 1998-2018 by the Free Software Foundation, Inc.
 #
diff -urNp -x '*.orig' mailman-2.1.34.org/cron/senddigests mailman-2.1.34/cron/senddigests
--- mailman-2.1.34.org/cron/senddigests	2020-06-27 02:21:21.000000000 +0200
+++ mailman-2.1.34/cron/senddigests	2021-10-23 13:32:45.963703728 +0200
@@ -1,4 +1,4 @@
-#! @PYTHON@
+#! @PYTHON@ -S
 #
 # Copyright (C) 1998-2018 by the Free Software Foundation, Inc.
 #
