diff -ur mailman-2.0.13.orig/configure.in mailman-2.0.13/configure.in
--- mailman-2.0.13.orig/configure.in	Thu Aug 22 23:26:54 2002
+++ mailman-2.0.13/configure.in	Thu Aug 22 23:29:17 2002
@@ -240,11 +240,7 @@
 MM_FIND_USER_ID(MAILMAN_UID, $USERNAME)
 if test -z "$MAILMAN_UID"
 then
-    AC_MSG_ERROR([
-***** No \"$USERNAME\" user found!
-***** Your system must have a \"$USERNAME\" user defined
-***** (usually in your /etc/passwd file).  Please see the INSTALL
-***** file for details.])
+MAILMAN_UID="94"
 fi
 
 
@@ -267,11 +263,7 @@
 MM_FIND_GROUP_ID(MAILMAN_GID, $GROUPNAME)
 if test -z "$MAILMAN_GID"
 then
-    AC_MSG_ERROR([
-***** No \"$GROUPNAME\" group found!
-***** Your system must have a \"$GROUPNAME\" group defined
-***** (usually in your /etc/group file).  Please see the INSTALL
-***** file for details.])
+MAILMAN_GID="94"
 fi
 
 
@@ -368,12 +360,7 @@
 MM_FIND_GROUP_ID(CGI_GID, $ac_cv_group_cgi)
 if test -z "$CGI_GID"
 then
-    AC_MSG_ERROR([
-***** No existing group found for the cgi_wrapper program.
-***** This is the group that your Web server runs CGI scripts under.
-***** You might want to specify an existing group with the
-***** --with-cgi-gid configure option.  Please see your Web server's
-***** documentation, and the INSTALL file for details])
+CGI_GID=51
 fi
 
 
@@ -394,80 +381,6 @@
 fi
 AC_MSG_RESULT($with_cgi_ext)
 
-
-# figure out the DEFAULT_HOST_NAME and DEFAULT_URL
-AC_SUBST(FQDN)
-AC_SUBST(URL)
-changequote(,)
-cat > conftest.py <<EOF
-# attempt to figure out the default hostname and URL
-from socket import *
-import string
-import sys
-import os
-def barf(fqdn, www):
-    sys.stdout = sys.stderr
-    print 'host info not found, set \$MAILHOST and/or \$WWWHOST environ vars'
-    print '\$MAILHOST=%s, \$WWWHOST=%s' % (fqdn, www)
-    sys.exit(1)
-fqdn = os.environ.get('MAILHOST')
-www = os.environ.get('WWWHOST')
-aliases = []
-if fqdn:
-    aliases.append(fqdn)
-if www:
-    aliases.append(www)
-if not fqdn:
-    try:
-        host, aliases, ipaddrs = gethostbyaddr(gethostbyname(gethostname()))
-    except error:
-        barf(fqdn, www)
-    aliases.insert(0, host)
-for h in aliases:
-    parts = string.split(h, '.')
-    if len(parts) > 1:
-        if parts[0] == 'www':
-            www = h
-        elif not fqdn:
-            fqdn = h
-    if www and fqdn:
-        break
-if fqdn is None:
-    barf(fqdn, www)
-if www is None:
-    www = fqdn
-fp = open('conftest.out', 'w')
-if www and fqdn:
-    fp.write('%s\n%s\n' % (fqdn, www))
-elif not www and fqdn:
-    fp.write('%s\n%s\n' % (fqdn, fqdn))
-elif www:
-    dhn = string.join(string.split(www, '.')[1:], '.')
-    fp.write('%s\n%s\n' % (dhn, www))
-else:
-    fp.write('please.change.me\nwww.please.change.me\n')
-fp.close()
-EOF
-$PYTHON conftest.py
-if [ $? -eq 1 ]
-then
-    exit
-fi
-changequote([, ])
-AC_MSG_CHECKING(for default fully qualified host name)
-if test -z "$FQDN"
-then
-    FQDN=`head -1 conftest.out`
-fi
-AC_MSG_RESULT($FQDN)
-AC_MSG_CHECKING(for default URL host component)
-if test -z "$URL"
-then
-    URL=`tail -1 conftest.out`
-fi
-AC_MSG_RESULT($URL)
-rm -f conftest.out conftest.py
-
 # Checks for libraries.
 AC_CHECK_FUNCS(strerror setregid syslog)
 if test $ac_cv_func_syslog = no; then
Only in mailman-2.0.13: configure.in.orig
Only in mailman-2.0.13: configure.in.rej
Only in mailman-2.0.13: configure.in~
diff -ur mailman-2.0.13.orig/templates/Makefile.in mailman-2.0.13/templates/Makefile.in
--- mailman-2.0.13.orig/templates/Makefile.in	Thu Aug 22 23:26:54 2002
+++ mailman-2.0.13/templates/Makefile.in	Thu Aug 22 23:27:35 2002
@@ -59,7 +59,6 @@
 install:
 	for f in $(TEMPLATES); \
 	do \
-	    mv -f $(TEMPLATEDIR)/$$f $(TEMPLATEDIR)/$$f.prev || $(TRUE); \
 	    $(INSTALL) -m $(FILEMODE) $$f $(TEMPLATEDIR); \
 	done
 
