diff -urN mailman-2.0.2.org/configure.in mailman-2.0.2/configure.in
--- mailman-2.0.2.org/configure.in	Sun Mar  4 10:34:36 2001
+++ mailman-2.0.2/configure.in	Sun Mar  4 19:26:52 2001
@@ -234,11 +234,7 @@
 MM_FIND_USER_ID(MAILMAN_UID, $USERNAME)
 if test -z "$MAILMAN_UID"
 then
-    AC_MSG_ERROR([
-***** No \"$USERNAME\" user found!
-***** Your system must have a \"$USERNAME\" user defined
-***** (usually in your /etc/passwd file).  Please see the INSTALL
-***** file for details.])
+MAILMAN_UID="94"
 fi
 
 
@@ -261,11 +257,7 @@
 MM_FIND_GROUP_ID(MAILMAN_GID, $GROUPNAME)
 if test -z "$MAILMAN_GID"
 then
-    AC_MSG_ERROR([
-***** No \"$GROUPNAME\" group found!
-***** Your system must have a \"$GROUPNAME\" group defined
-***** (usually in your /etc/group file).  Please see the INSTALL
-***** file for details.])
+MAILMAN_GID="94"
 fi
 
 
@@ -280,49 +272,6 @@
     prefixcheck=$VAR_PREFIX
 fi
 
-AC_MSG_CHECKING(permissions on $prefixcheck)
-changequote(,)
-cat > conftest.py <<EOF
-import os, grp, string
-from stat import *
-prefix = "$prefixcheck"
-mailmangid = $MAILMAN_GID
-problems = []
-try: statdata = os.stat(prefix)
-except os.error:
-    problems.append("Directory doesn't exist: " + prefix)
-else:    
-    mode = statdata[ST_MODE]
-    gid = statdata[ST_GID]
-    if mailmangid <> gid:
-	problems.append("Directory must be owned by group mailman: " + prefix)
-    if (mode & S_ISGID) <> S_ISGID:
-	problems.append("Set-gid bit must be set for directory: " +prefix)
-    perms = S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH
-    if (mode & perms) <> perms:
-	problems.append("Permissions should be at least 02775: " + prefix)
-if not problems:
-    msg = "okay\n"
-else:
-    msg = '***** ' + string.join(problems, '\n***** ') + '\n'
-fp = open("conftest.out", "w")
-fp.write(msg)
-fp.close()
-EOF
-changequote([, ])
-$PYTHON conftest.py
-status=`cat conftest.out`
-rm -f conftest.out conftest.py
-if test "$status" != "okay"
-then
-    AC_MSG_ERROR([
-***** Installation directory $prefixcheck is not configured properly!
-$status])
-fi
-
-AC_MSG_RESULT(okay)
-
-
 # Now find the UIDs and GIDs
 # Support --with-mail-gid and --with-cgi-gid
 AC_MSG_CHECKING(for mail wrapper GID)
@@ -362,12 +311,7 @@
 MM_FIND_GROUP_ID(CGI_GID, $ac_cv_group_cgi)
 if test -z "$CGI_GID"
 then
-    AC_MSG_ERROR([
-***** No existing group found for the cgi_wrapper program.
-***** This is the group that your Web server runs CGI scripts under.
-***** You might want to specify an existing group with the
-***** --with-cgi-gid configure option.  Please see your Web server's
-***** documentation, and the INSTALL file for details])
+CGI_GID=51
 fi
 
 
diff -urN mailman-2.0.2.org/templates/Makefile.in mailman-2.0.2/templates/Makefile.in
--- mailman-2.0.2.org/templates/Makefile.in	Sun Mar  4 10:34:36 2001
+++ mailman-2.0.2/templates/Makefile.in	Sun Mar  4 19:31:19 2001
@@ -59,7 +59,6 @@
 install:
 	for f in $(TEMPLATES); \
 	do \
-	    mv -f $(TEMPLATEDIR)/$$f $(TEMPLATEDIR)/$$f.prev || $(TRUE); \
 	    $(INSTALL) -m $(FILEMODE) $$f $(TEMPLATEDIR); \
 	done
 
diff -uNr mailman-2.0.12/configure.in mailman-2.0.12..new/configure.in
--- mailman-2.0.12/configure.in	Tue Aug  6 18:29:52 2002
+++ mailman-2.0.12..new/configure.in	Tue Aug  6 18:29:10 2002
@@ -332,78 +332,6 @@
 fi
 AC_MSG_RESULT($with_cgi_ext)
 
-
-# figure out the DEFAULT_HOST_NAME and DEFAULT_URL
-AC_SUBST(FQDN)
-AC_SUBST(URL)
-changequote(,)
-cat > conftest.py <<EOF
-# attempt to figure out the default hostname and URL
-from socket import *
-import string
-import sys
-import os
-def barf(fqdn, www):
-    sys.stdout = sys.stderr
-    print 'host info not found, set \$MAILHOST and/or \$WWWHOST environ vars'
-    print '\$MAILHOST=%s, \$WWWHOST=%s' % (fqdn, www)
-    sys.exit(1)
-fqdn = os.environ.get('MAILHOST')
-www = os.environ.get('WWWHOST')
-aliases = []
-if fqdn:
-    aliases.append(fqdn)
-if www:
-    aliases.append(www)
-if not fqdn:
-    try:
-        host, aliases, ipaddrs = gethostbyaddr(gethostbyname(gethostname()))
-    except herror:
-        barf(fqdn, www)
-    aliases.insert(0, host)
-for h in aliases:
-    parts = string.split(h, '.')
-    if len(parts) > 1:
-        if parts[0] == 'www':
-            www = h
-        elif not fqdn:
-            fqdn = h
-    if www and fqdn:
-        break
-if fqdn is None:
-    barf(fqdn, www)
-if www is None:
-    www = fqdn
-fp = open('conftest.out', 'w')
-if not www and fqdn:
-    fp.write('%s\n%s\n' % (fqdn, fqdn))
-elif www:
-    dhn = string.join(string.split(www, '.')[1:], '.')
-    fp.write('%s\n%s\n' % (dhn, www))
-else:
-    fp.write('please.change.me\nwww.please.change.me\n')
-fp.close()
-EOF
-$PYTHON conftest.py
-if [ "$?" == "1" ]
-then
-    exit
-fi
-changequote([, ])
-AC_MSG_CHECKING(for default fully qualified host name)
-if test -z "$FQDN"
-then
-    FQDN=`head -1 conftest.out`
-fi
-AC_MSG_RESULT($FQDN)
-AC_MSG_CHECKING(for default URL host component)
-if test -z "$URL"
-then
-    URL=`tail -1 conftest.out`
-fi
-AC_MSG_RESULT($URL)
-rm -f conftest.out conftest.py
-
 # Checks for libraries.
 AC_CHECK_FUNCS(strerror setregid syslog)
 if test $ac_cv_func_syslog = no; then
